src/openstack_cloud/openstack_manager.py

```diff
@@ -1509,20 +1520,71 @@ def _issue_reconciliation_metric(
         except IssueMetricEventError:
             logger.exception("Failed to issue Reconciliation metric")
 
-    def flush(self) -> int:
+    def flush(self, mode: FlushMode = FlushMode.FLUSH_IDLE) -> int:
         """Flush Openstack servers.
 
+        1. Kill the processes depending on flush mode.
+        2. Get unhealthy runners after process purging.
+        3. Delete unhealthy runners.
```

Comment:
Why not just delete the machine in OpenStack / LXD?

Reply:
There are follow-up processes such as metrics collection, key deletion. To reduce the responsibility of the function, the flush method will only handle deletion of the processes and let the normal flow continue.